Move escaping from Attribute constructor to rendering

`Attribute` escapes its value in the constructor, that leads to the behaviour that `Attribute.getValue()` returns escaped text.

If that text is then used to create another value or text element, it will be double escaped `&lt;` -> `&amp;lt;`

```
    @Test
    public void testAttributeValueRoundtrip() throws Exception {
        Attribute attributeWithValue = new Attribute("href", "<");
        Attribute attributeWithValueReconstructed = new Attribute("href", attributeWithValue.getValue());

        assertThat(attributeWithValueReconstructed.getValue(), is("<"));
    }
```


```
java.lang.AssertionError: 
Expected: is "<"
     but: was "&amp;lt;"
Expected :<
Actual   :&amp;lt;
```


Additionally the behaviour is inconsistent between the constructor and the `setValue` method. `setValue()` does not escape the text, while the constructor does.

Changing this will technically alter behaviour of existing code, but one could argue that uses like this were broken.

--- a/src/main/java/j2html/attributes/Attribute.java
+++ b/src/main/java/j2html/attributes/Attribute.java
@@ -10,7 +10,7 @@ public class Attribute implements Renderable {
 
     public Attribute(String name, String value) {
         this.name = name;
-        this.value = Config.textEscaper.escape(value);
+        this.value = value;
     }
 
     public Attribute(String name) {
@@ -23,12 +23,12 @@ public class Attribute implements Renderable {
         if (name == null) {
             return;
         }
-        writer.append(" ");
+        writer.append(' ');
         writer.append(name);
         if (value != null) {
             writer.append("=\"");
-            writer.append(value);
-            writer.append("\"");
+            writer.append(Config.textEscaper.escape(value));
+            writer.append('"');
         }
     }
 

--- a/src/main/java/j2html/utils/EscapeUtil.java
+++ b/src/main/java/j2html/utils/EscapeUtil.java
@@ -6,7 +6,7 @@ public class EscapeUtil {
         if (s == null) {
             return null;
         }
-        StringBuilder escapedText = new StringBuilder();
+        StringBuilder escapedText = new StringBuilder(s.length()+16);
         char currentChar;
         for (int i = 0; i < s.length(); i++) {
             currentChar = s.charAt(i);

